.github/workflows/build_windows_artifacts.yml

```diff
@@ -165,12 +175,66 @@ jobs:
             echo "CCache Stats:"
             echo "-------------"
             ccache -s
+
+            # Collect artifact info
+            artifact_list=$(ls -lh "${{ env.BUILD_DIR }}"/artifacts/*.tar.xz 2>/dev/null | awk '{print $9 " (" $5 ")"}' | jq -R . | jq -s .)
+            artifact_sizes=$(du -h -d 1 "${{ env.BUILD_DIR }}/artifacts" 2>/dev/null | awk '{print $2 ": " $1}' | jq -R . | jq -s .)
+            sdk_sizes=$(du -h -d 1 "${{ env.BUILD_DIR }}/dist/rocm" 2>/dev/null | awk '{print $2 ": " $1}' | jq -R . | jq -s .)
+
+            # Capture ccache stats
+            ccache_stats=$(ccache -s | jq -R -s .)
+
+            # Determine build outcome
+            build_outcome="success"
           else
             echo "[ERROR] Build directory ${{ env.BUILD_DIR }} does not exist. Skipping report!"
-            echo "        This should only happen if the CI is cancelled before the build step."
-            exit 1
+            echo "[ERROR] This should only happen if the CI is cancelled before the build step."
+            build_outcome="failed"
+            mkdir -p build/logs
+            echo "[ERROR] Build directory missing" > build/logs/error.log
           fi
 
+          # Build JSON file (always create it)
+          jq -n \
+            --arg workflow "${{ github.workflow }}" \
+            --arg run_id "${{ github.run_id }}" \
+            --arg run_number "${{ github.run_number }}" \
+            --arg commit "${{ github.sha }}" \
+            --arg repo "${{ github.repository }}" \
+            --arg timestamp "$(date -u '+%Y-%m-%dT%H:%M:%SZ')" \
+            --arg hostname "$HOSTNAME" \
+            --arg build_outcome "$build_outcome" \
+            --argjson artifacts "${artifact_list:-[]}" \
+            --argjson artifact_sizes "${artifact_sizes:-[]}" \
+            --argjson sdk_sizes "${sdk_sizes:-[]}" \
+            --arg ccache_stats "${ccache_stats:-""}" \
+            '{
+              workflow: $workflow,
+              run_id: $run_id,
+              run_number: $run_number,
+              commit: $commit,
+              repo: $repo,
+              timestamp: $timestamp,
+              hostname: $hostname,
+              build_outcome: $build_outcome,
+              build_report: {
+                artifacts: $artifacts,
+                artifact_sizes: $artifact_sizes,
+                sdk_sizes: $sdk_sizes,
+                ccache_stats: $ccache_stats
+              }
+            }' > build/logs/build_report_${HOSTNAME}.json
+
+          echo "Build report saved to build/logs/build_report_${HOSTNAME}.json"
```

Comment:
To me it seems that all this needs to go to a Python script as all this logic is way beyond what should be included in a workflow step.